extensions/interactions/GraphInput/GraphInput.js

```diff
@@ -334,15 +335,19 @@ oppia.directive('graphViz', function() {
           }
           $scope.graph[option] = !$scope.graph[option];
         };
-        $scope.setMode = function(mode, $event) {
-          // Prevents new vertex from being added in add edge mode
-          $event.preventDefault();
-          $event.stopPropagation();
+
+        var setMode = function(mode) {
           $scope.state.currentMode = mode;
           $scope.state.addEdgeVertex = null;
           $scope.state.selectedVertex = null;
           $scope.state.selectedEdge = null;
         };
+        $scope.onClickModeButton = function(mode, $event) {
+          // Prevents new vertex from being added in add edge mode
```

Comment:
I feel that would be a little strange, since setMode() accounts for all modes rather than just the case of adding. I think it would be best to just remove the comment--I think the functionality of the code is straightforward enough without it.